Dockerfile.template

```diff
@@ -32,3 +32,11 @@ RUN curl -sSL "https://github.com/gotestyourself/gotestsum/releases/download/v${
 
 ENV GOPATH /home/circleci/go
 ENV PATH $GOPATH/bin:/usr/local/go/bin:$PATH
+
+# Install govulncheck on a temporary stage to ensure minimal size footprint
+# by copying to output layer just the govulncheck binary
+FROM base AS govulncheck
+RUN go install "golang.org/x/vuln/cmd/govulncheck@latest"
```

Comment:
This should be decided by @JalexChen I believe but I think we'd want to pin the version here and then add it to the updated script.

Reply:
Will address your comments by applying suggestions once JalexChen provides feedback on version pinning

Reply:
Addressed on https://github.com/CircleCI-Public/cimg-go/pull/242/commits/871a7192857023af1430ea47e9d007817d480a68

Reply:
i apologize -  i have not seen these tags for some reason, but i think this ended up being resolved in a later commit. i think the idea of having a version we can pin and is visible, rather than @latest is helpful for everyone if there needs to be debugging later on